test/unit/lodash.spec.js

```diff
@@ -0,0 +1,690 @@
+            var params = datasets.partition3(users);
+            it('should match types of passing elements', function () {
+                expect(typeof params[1]).to.be.oneOf(['number', 'string', 'object', 'boolean']);
+            });
+            it('should split persons only to falsy table when predicate is false for all', function () {
+                expect(_.partition.apply(_, datasets.partition3(users))).to.eql([[],
+                    [
+                        {name: 'Andrinio', age: 36, active: false},
+                        {name: 'Harry', age: 40, active: true},
+                        {name: 'Kodi', age: 1, active: false},
+                        {name: 'Francheska', age: 61, active: true}
+                    ]
+                ]);
+            });
+        });
+    });
+
+    describe('reduce', function () {
+        var params = datasets.reduce();
+        var array = params[0];
+        var spy = sinon.spy(params[1]);
+
+        it('should match types of passing elements', function () {
+            var element1 = params[0] instanceof Array;
+            var element2 = params[1] instanceof Function;
+            var element3 = typeof params[2];
+
+            expect(element1).to.eql(true);
+            expect(element2).to.eql(true);
+            expect(element3).to.eql('number')
+            expect(params.length).to.eql(3);
+        });
+        it('should pass mock test', function () {
+            var testArray = [10, 11, 12, 15];
+            var accumulator = 0;
+            var checkFunc = _.reduce.apply(_, [testArray, spy, accumulator]);
+            expect(spy).to.have.been.callCount(4);
+            expect(checkFunc).to.eql(590);
+        });
+        it('should count sum of squared numbers', function () {
+            array.forEach(function (e) {
+                expect(typeof e).to.eql('number');
+            });
+            expect(array.length).to.eql(5);
+            expect(array).to.include(1);
+            expect(array).to.include(5);
+            expect(_.reduce.apply(_, datasets.reduce())).to.eql(55);
+        });
+    });
+
+    describe('sortBy', function () {
+        var params = datasets.sortBy();
+        var array = params[0];
+
+        it('should match types of passing elements', function () {
+            var element1 = params[0] instanceof Array;
+            var element2 = typeof params[1];
+
+            expect(element1).to.eql(true);
+            expect(element2).to.eql('string');
+            expect(params.length).to.eql(2);
+        });
+        it('should contain array of random strings with different lenghts', function () {
+            var checkLetters = array[0].substr(0, 3);
+            var checkLetters2 = array[2].substr(2, 5);
+            var checkLetters3 = array[3].substr(6, 10);
+            var checkLetters4 = array[4].substr(3, 6);
+
+            expect(array).to.have.length(6);
+            array.forEach(function (item, index) {
```

Comment:
Absolutely too crazy logic for this exercise. It took me long time to make it work and I hoped it would mean something for the exercise but it didn't. To be honest I was absolutely dissapointed and felt like I've wasted my time. Simplify it and remake this with more focus on how it should be checked, maybe not length property but something more creative?